helpers/store/composer.js

```diff
@@ -1,43 +1,89 @@
+import commonHelper from '~/helpers/common'
+import knitLogger from '~/config/logger'
 import _ from 'lodash'
 
-const methods = {
-  cancel () {
-    throw new Error('cancel')
+function _validateComposableInputs (name, caller) {
+  if (!name) {
+    throw Error('Action name should be specified')
+  }
+
+  if (!caller) {
+    throw Error('Caller must be defined')
   }
 }
 
-function compose ({ before, caller, success, always, fail }) {
-  // Binds all parameters to all success, always functions
-  const promisesBoundCaller = async (arrFn, that, ctx, params, result) => {
-    const boundPromises = _.map(arrFn, async (bind) => bind({ that, ctx, params, result }))
+// Binds all parameters to all success, always functions
+const _promisesBoundCaller = async (arrFn, that, ctx, params, result) => {
+  const bound = { that, ctx, params, result }
+  const boundPromises = _.map(arrFn, async bind => bind(bound))
 
-    for (const promise of boundPromises) {
-      await promise
-    }
+  for (const promise of boundPromises) {
+    await promise
   }
+}
+// Binds all parameters to the action
+const _actionWithBoundCaller = (that, ctx, params, actionDefn) =>
+  () => actionDefn.bind(that)(ctx, params)
 
-  // Binds all parameters to the action
-  const actionWithBoundCaller = (that, ctx, params, actionDefn) =>
-    () => actionDefn.bind(that)(ctx, params)
+// Binds all parameters to all composers, caller
+const _flowWithBoundCaller = (before, caller, that, ctx, params) => {
```

Comment:
`that`, `ctx`, `params` are duplicated in `_actionWithBoundCaller` and `_flowWithBoundCaller` maybe we should think to be consistent and have them in the same order in declaration?

i propose:

'const _actionWithBoundCaller = (that, ctx, params, actionDefn)`
and
`const _flowWithBoundCaller = (that, ctx, params, before, caller)`